### DISCUSSION TITLE
Can we render html within pdf file?

### DISCUSSION THREAD

**Discussion by nipnipj** [quarto-dev/quarto-cli#5921] [Q&A]
### Description

My gt tables works fine when rendering a .qmd document into .html. Everything is ok --- no errors or warnings. When I try to render the same .qmd document into .pdf, tables lose colors and `Warning: HTML tags found, and they will be removed.` message appears. Is there a way to render (not using images) these tables within a .pdf file or is just not possible.

**Comment by cscheid** [comment]
From what I understand it, the extensive styling allowed in GT is only supported in HTML. @rich-iannone can you confirm?

**Reply by rich-iannone** [reply]
Yes, LaTeX tables have a bit more catching up to do and most of that centers around styles and column widths. I am a bit surprised about that warning you got. It’s from gt and it happens when you try use HTML tags within any labels. Would you mind sharing the code? Then we can go from there.

**Reply by nipnipj** [reply]
This reproduce the warning. It appears when `cols_label()` is used.
```
tibble(
  Age_group = c("<35", "35-44", "45-54", "55-64", "65-74", md(">=75")),
  Cases_nij_Southern = c(64,75,68,63,45,27),
  Estimated_northern = c(2880262, 564535, 592983, 450740, 270908, 161850),
  Population_at_risk_Nij_Southern = c(1074246, 220407, 198119, 134084, 70708, 34233)
) %>% 
  gt() %>%
   cols_label(
    Age_group = md("**Age group**"),
    Cases_nij_Southern = md("**Cases, n<sub>ij</sub> <br /> Southern**"),
    Estimated_northern = md("**Estimated <br /> Northern**"),
    Population_at_risk_Nij_Southern = md("**<center>Population at risk, 
                                         N<sub>ij</sub> <br /> Southern</center>**")
  ) %>% 
  data_color(columns = c(Cases_nij_Southern),
             apply_to = "fill",
             colors = scales::col_numeric(
               palette = c("red", "orange", "blue"),
               domain = NULL)) %>% 
  tab_style(
    style = list(
      cell_fill(color = "red")
      #cell_text(style = "italic")
    ),
    locations = cells_body(columns = 1, rows = 1)
  )
```
The colors don't work for .pdf formats.

**Reply by rich-iannone** [reply]
Thanks for sharing the example. If you’re rendering a table to LaTeX you can’t have HTML tags within an `md()` call (fine for HTML tables though). The gt package will warn you (but not error: https://github.com/rstudio/gt/blob/4f9b9fe23aeba4f90609019c535c2c19181dee11/R/utils.R#L684). Background fills and text colors are not yet possible for LaTeX tables. Once I’m working on the package again, I’ll see to improving LaTeX output in this way.
